test/blackbox-tests/test-cases/pkg/opam-package-with-subst.t

```diff
@@ -0,0 +1,54 @@
+We test how opam files with substs fields are translated into the dune.lock file.
+ 
+  $ . ./helpers.sh
+
+Generate a mock opam repository
+  $ mkdir -p mock-opam-repository
+  $ cat >mock-opam-repository/repo <<EOF
+  > opam-version: "2.0"
+  > EOF
+
+Make a package with a substs field 
+  $ mkpkg with-substs <<EOF
+  > opam-version: "2.0"
+  > substs: ["foo.ml"]
+  > build: [ ["[" "-e" "foo.ml" "]"] [ "cat" "foo.ml"] ]
+  > EOF
+
+  $ opam_repo=mock-opam-repository/packages/with-substs/with-substs.0.0.1
+
+  $ solve_project <<EOF
+  > (lang dune 3.8)
+  > (package
+  >  (name x)
+  >  (allow_empty)
+  >  (depends with-substs)) 
+  > EOF
+  Solution for dune.lock:
+  with-substs.0.0.1
+  
+  $ cat >>dune.lock/with-substs.pkg <<EOF
+  > (source (copy $PWD/source))
+  > EOF
+
+The lockfile should contain the substitute action. The generation step currently doesn't
+add this in.
+
+  $ cat dune.lock/with-substs.pkg 
+  (version 0.0.1)
+  
+  (build
+   (run [ -e foo.ml ]))
```

Comment:
By the way, this translation step seems like a bug. This should be a system action

Reply:
Opened #8663.

Reply:
The bug is in your opam file. It should use `sh -c` to execute the shell command.

Reply:
Fixed.